contract.md

```diff
@@ -4,6 +4,8 @@
 
 _At a high level, what do you most want to get out of this project, and why?_
 
+	What I most want to get out of this project is a better understanding of how to effectively design a project? Since this is a class on Domain Specific Languages, I'm obviously going to be doing so through the lens of designing a custom programming language to fit a specific domain. One of the big issues I have with working on projects outside this course is wanting to get right into solving the technical issues that the project presents, which can often lead to unintuitive and sometimes counterproductive design choices being made in the middle of the project. In this project, I kind of want the experience of going in with a specific plan for what to do, and then making deliberate changes across the project as hurdles are found.
```

Comment:
I also have this issue sometimes; instead of focusing on the design, I focus on the implementation which is arguably worse because it doesn't matter how good your code is if it's not intuitive for the users.